arrow-libs/resilience/arrow-resilience/src/commonMain/kotlin/arrow/resilience/Schedule.kt

```diff
@@ -431,21 +455,37 @@ public suspend fun <Input, Output> Schedule<Throwable, Output>.retryOrElse(
 public suspend fun <Input, Output, A> Schedule<Throwable, Output>.retryOrElseEither(
   action: suspend () -> Input,
   orElse: suspend (Throwable, Output) -> A
+): Either<A, Input> =
+  retryOrElseEither<Input, Output, A, Throwable>(action, orElse)
+
+/**
+ * Retries [action] only when [E] has occurred, otherwise it immediately retries the other exception.
+ * If the [Schedule] is exhausted,
+ * it will invoke [orElse] with the last [E] and the output of the [Schedule] to produce a fallback value of [A].
+ * Returns [Either] with the fallback value if the [Schedule] is exhausted, or the successful result of [action].
+ */
+@JvmName("retryOrElseEitherReified")
+public suspend inline fun <Input, Output, A, reified E : Throwable> Schedule<E, Output>.retryOrElseEither(
```

Comment:
We should strive to keep the `inline` functions to be just small wrappers over the real functions (otherwise every change in the implementation breaks binary compatibility). My suggestion is to have a version which takes a `KClass<E>` as additional parameter (so you don't need the `reified` anymore), and then have a version which simply calls that function with `E::class`.